### ISSUE TITLE
 Issue with has_training_support API on AArch64 for FP16

### ISSUE THREAD

**Issue by cfRod** [oneapi-src/oneDNN#1510]
# Summary
Commit https://github.com/oneapi-src/oneDNN/commit/64198427cc3a84fb15e19691cf6477dbe1c4a3d0 is causing an issue on AArch64 . The failing test appears for test_batch_normalization/ BatchNormalizationSimpleF16 cases which are calling into reference batch normalisation kernels for oneDNN builds with ACL. 

# Version
https://github.com/oneapi-src/oneDNN/commit/64198427cc3a84fb15e19691cf6477dbe1c4a3d0 

# Environment
CPU aarch64
OS version Ubuntu 20.04.4
Compiler version 10.3.0
CMake version 3.24.1

# Steps to reproduce
```
./tests/gtests/test_batch_normalization  --gtest_filter="BatchNormalizationSimpleF16*" 

```

# Observed behavior
This calls into ref/ncsp/nscp implementation and returns false for  has_training_support https://github.com/oneapi-src/oneDNN/blob/b4b2d5265ff974779cc7ed2ba878600524366eba/src/cpu/platform.cpp#L154 and this causes the ctest to fail with it not being able to create a primitive descriptor for this case.
```
unknown file: Failure
C++ exception with description "could not create a primitive descriptor for a batch normalization forward propagation primitive" thrown in SetUp().
[  FAILED  ] BatchNormalizationSimpleF16/batch_normalization_test_t.TestsBatchNormalization/4, where GetParam() = 56-byte object <01-00 00-00 01-00 00-00 00-00 00-00 05-00 00-00 05-00 00-00 05-00 00-00 60-37 3D-E2 AA-AA 00-00 80-37 3D-E2 AA-AA 00-00 80-37 3D-E2 AA-AA 00-00 00-00 00-00 00-00 00-00> (8 ms)
[ RUN      ] BatchNormalizationSimpleF16/batch_normalization_test_t.TestsBatchNormalization/5
```
[bnorm-ctest.log](https://github.com/oneapi-src/oneDNN/files/10094246/bnorm-ctest.log)

ACL has FP16 support and we have added a check for it to (has_data_type_support) https://github.com/oneapi-src/oneDNN/blob/b4b2d5265ff974779cc7ed2ba878600524366eba/src/cpu/platform.cpp#L127 but it does not have support for training in FP16. 

Should the whole test be skipped on non-x86 builds? For e.g by adding a check here https://github.com/oneapi-src/oneDNN/blob/b4b2d5265ff974779cc7ed2ba878600524366eba/tests/gtests/test_batch_normalization.cpp#L63? 

Also, what is the difference of has_training_support and has_data_type_support? Could you clarify this? Is it just some hardware supports FP16 on Intel?

**Comment by dzarukin** [comment]
Hi @cfRod, thank you for reporting an issue.

Hi @xuxinzen, could you, please, take a look on this one? Thank you.

**Comment by vpirogov** [comment]
Fixed by e535ef2f8cbfbee4d385153befe508c6b054305e.

**Comment by nSircombe** [comment]
Thanks @vpirogov, that looks to be working for me!

**Comment by cfRod** [comment]
Thanks @vpirogov @xuxinzen ,

Can you clarify if this check means whether the underlying library (ACL) has to have training support or whether the underlying library can train the model using datatype of fp16 or whether the hardware supports fp16.

The fix https://github.com/oneapi-src/oneDNN/commit/e535ef2f8cbfbee4d385153befe508c6b054305e is a bit misleading because 
1) ACL does not support training
2) ACL does not support training with fp16
3) ACL does support fp16 and arm_compute::CPUInfo::get().has_fp16() is checking whether the hardware has fp16 datatype. 

What is the different between has_training_support and has_data_type_support?

**Comment by xuxinzen** [comment]
> Thanks @vpirogov @xuxinzen ,
> 
> Can you clarify if this check means whether the underlying library (ACL) has to have training support or whether the underlying library can train the model using datatype of fp16 or whether the hardware supports fp16.
> 
> The fix [e535ef2](https://github.com/oneapi-src/oneDNN/commit/e535ef2f8cbfbee4d385153befe508c6b054305e) is a bit misleading because
> 
> 1. ACL does not support training
> 2. ACL does not support training with fp16
> 3. ACL does support fp16 and arm_compute::CPUInfo::get().has_fp16() is checking whether the hardware has fp16 datatype.
> 
> What is the different between has_training_support and has_data_type_support?

Sorry for the late response.
1. `has_data_type_support()` is intend to check whether the hardware supports fp16(or any specific data tpye).
2. `has_training_support()` is intend to check whether ACL supports training with fp16(or any specific data type).

During extension support of future hardware, we plan to only enable inference for fp16/bf16 for now even it supports fp16/bf16 in hardware level. So `has_training_support()` helps us to check about it.

**Comment by xuxinzen** [comment]
Ok, we can correct `has_training_support()`. And this issue should be fix by cb12bc594a6c14914abeca3b86b82c3d5edaaa4e after we revert the change for ACL.

**Comment by xuxinzen** [comment]
We also disable ref implementation for training with fp16/bf16. Does ACL support ref impl for training with fp16/bf16?

**Comment by cfRod** [comment]
Hi ACL, does not have reference implementations. We rely on oneDNN's reference implementation for AArch64. Not sure if that is what you mean?
